docs/support/tutorials/nextflow-puhti.md

```diff
@@ -1,191 +1,309 @@
 #SBATCH --partition=small
-#SBATCH --account=project_XXX
+#SBATCH --account=<project>
 #SBATCH --cpus-per-task=4
 
-export TMPDIR=$PWD
-
-# Activate  Nextflow on Puhti
+export TMPDIR=${PWD}
+export SCRATCH=/scratch/<project>/What_the_Phage
 
+# Activate Nextflow on Puhti
 module load nextflow/21.10.6 
 
 # Nextflow command here
-nextflow run /scratch/project_xxx/What_the_Phage/phage.nf --fasta /scratch/project_xxx/What_the_Phage/test-data/OX2_draft.fa --cores 4 --output results -profile local,singularity --cachedir /scratch/project_xxx/What_the_Phage/singularity --databases /scratch/project_xxx/What_the_Phage/databases/WtP_databases --workdir /scratch/project_xxx/What_the_Phage/workflow-phages-username 
+nextflow run ${SCRATCH}/phage.nf \
+         --fasta ${SCRATCH}/test-data/OX2_draft.fa \
+         --cores 4 \
+         --output results \
+         --profile local,singularity \
+         --cachedir ${SCRATCH}/singularity \
+         --databases ${SCRATCH}/databases/WtP_databases \
+         --workdir ${SCRATCH}/workflow-phages-username
+```
+
+Please note that the container images in this example should be in the folder
+`/scratch/<project>/What_the_Phage/singularity`. Otherwise, Nextflow tries to
+download the images again which increases the likelihood of failures. Remember
+to edit `<project>` to match your actual CSC computing project name.
+
+## 7. Demonstration of nf-core Nextflow pipeline using HyperQueue executor (optional)
+
+In this example, let's use the
+[HyperQueue meta-scheduler](../../apps/hyperqueue.md) for executing a Nextflow
+pipeline. This executor can be used to scale up analysis across multiple nodes
+when needed.
+
+Here is a batch script for running a nf-core pipeline on Puhti:
+
+```bash
+#!/bin/bash
+#SBATCH --job-name=nextflowjob
+#SBATCH --account=<project>
+#SBATCH --partition=small
+#SBATCH --time=00:10:00
+#SBATCH --nodes=1
+#SBATCH --ntasks-per-node=1
+#SBATCH --cpus-per-task=40
+#SBATCH --mem-per-cpu=2G
+  
+module load hyperqueue/0.16.0
+module load nextflow/22.10.1
+
+# create a directory on scratch folder for running nextflow pipeline
+export SCRATCH=/scratch/<project>/$USER/nextflow
+mkdir -p ${SCRATCH}
+cd ${SCRATCH}
+
+export APPTAINER_TMPDIR="${SCRATCH}"
+export APPTAINER_CACHEDIR="${SCRATCH}"
+unset XDG_RUNTIME_DIR
+
+# Specify a location for the HyperQueue server
+export HQ_SERVER_DIR=${PWD}/hq-server-${SLURM_JOB_ID}
+mkdir -p "${HQ_SERVER_DIR}"
+
+# Start the server in the background (&) and wait until it has started
+hq server start &
+until hq job list &>/dev/null ; do sleep 1 ; done
+
+# Start the workers in the background and wait for them to start
+srun --overlap --cpu-bind=none --mpi=none hq worker start --cpus=${SLURM_CPUS_PER_TASK} &
+hq worker wait "${SLURM_NTASKS}"
+
+# As an example, let's clone a nf-core pipeline and run a test sample
+git clone https://github.com/nf-core/rnaseq.git -b 3.10
+cd rnaseq
+nextflow run main.nf -profile test,singularity --outdir . -resume
```

Comment:
Should it be singularity or apptainer?

Reply:
for this one, it is "singularity".  In general, most nextflow pipelines still use singularity terminology. And also depends on the version of pipeline. Latest pipelines are slowly catching up with Apptainer terminology. I would use Apptainer and singularity interchangeably in the general description of Nextflow at least for now.